### PR TITLE
Node reboot and its impact on registry pods

### PR DESCRIPTION
**Pull request by Akarsha-rai**
Signed-off-by: Akarsha <akrai@redhat.com>

### CODE REVIEW

File: tests/e2e/registry/test_registry_reboot_node.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+                n for n in get_node_objs() if n
+                .ocp.get_resource_status(n.name) == NODE_NOT_READY
+            ]
+            log.warning(
+                f"Nodes in NotReady status found: {[n.name for n in not_ready_nodes]}"
+            )
+            if not_ready_nodes:
+                nodes.restart_nodes_by_stop_and_start(not_ready_nodes)
+                wait_for_nodes_status()
+            log.info("All nodes are in Ready status")
```
Comment (PrasadDesala) [inline tests/e2e/registry/test_registry_reboot_node.py:65]: Please use node_restart_teardown fixture here

File: tests/e2e/registry/test_registry_reboot_node.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+            # Clean up and remove namespace
+            log.info("Clean up and remove namespace")
+            ocp_obj.exec_oc_cmd(command=f'delete project {self.project_name}')
+
+            # Reset namespace to default
+            ocp.switch_to_default_rook_cluster_project()
+            ocp_obj.wait_for_delete(resource_name=self.project_name)
```
Comment (PrasadDesala) [inline tests/e2e/registry/test_registry_reboot_node.py:73]: project_factory_fixture does these things, can we use the fixture here?

File: tests/e2e/registry/test_registry_reboot_node.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+            delay=15)(
+            wait_for_nodes_status(timeout=900)
+        )
+
+        # Validate image registry pods
+        validate_registry_pod_status()
+
+        # Validate image exists in registries path
+        validate_image_exists(namespace=self.project_name)
+
```
Comment (PrasadDesala) [inline tests/e2e/registry/test_registry_reboot_node.py:129]: Also validate that all the storage pods are in running state post node recovery.

File: tests/e2e/registry/test_registry_reboot_node.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+                delay=15)(
+                wait_for_nodes_status(timeout=900)
+            )
+
+            # Validate image registry pods
+            validate_registry_pod_status()
+
+            # Validate image exists in registries path
+            validate_image_exists(namespace=self.project_name)
+
```
Comment (PrasadDesala) [inline tests/e2e/registry/test_registry_reboot_node.py:193]: Also validate that all the storage pods are in running state post node recovery.

File: tests/e2e/registry/test_registry_reboot_node.py (hunk 3 above)
Comment (Akarsha-rai) [reply tests/e2e/registry/test_registry_reboot_node.py:129]: below function sanity_helpers.health_check(), validates all nodes are in Ready state, all storage pods are in running and also ceph health is OK. Do we need to add another check for only storage pods ?

File: tests/e2e/registry/test_registry_reboot_node.py (hunk 4 above)
Comment (Akarsha-rai) [reply tests/e2e/registry/test_registry_reboot_node.py:193]: below function sanity_helpers.health_check(), validates all nodes are in Ready state, all storage pods are in running and also ceph health is OK. Do we need to add another check for only storage pods ?

File: tests/e2e/registry/test_registry_reboot_node.py (hunk 2 above)
Comment (Akarsha-rai) [reply tests/e2e/registry/test_registry_reboot_node.py]: done

File: tests/e2e/registry/test_registry_reboot_node.py (hunk 1 above)
Comment (Akarsha-rai) [reply tests/e2e/registry/test_registry_reboot_node.py]: done

**Comment by Akarsha-rai** [comment]
Job passed: https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/16367/testReport/

Screenshot from 2021-01-15 13-58-21

**Review by Shrivaibavi — APPROVED** [review]
LGTM

**Review by vavuthu — APPROVED** [review]

**Pull request merged**